src/main/java/br/org/soujava/pomeditor/mojo/EditingMojo.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2023  the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package br.org.soujava.pomeditor.mojo;
+
+import br.org.soujava.pomeditor.api.PomChange;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugins.annotations.Parameter;
+
+import java.nio.file.Path;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
```

Comment:
@veselov could you add a javadoc documentation for this abstract class explaining its purposes, please?

Reply:
Great!!! Thanks!